### PR TITLE
Add "New insight" button to new sidebar

### PR DESCRIPTION
**Pull request by Twixes**
## Changes

Oh how insightful it is to gain insight into your product's analytics by creating a new insight.
This revolutionary button allows you to do it when using the navigation redesign:

## How did you test this code?

Manual testing so far.

### CODE REVIEW

File: frontend/src/lib/components/LemonButton/index.tsx [hunk 1]
```diff
@@ -15,15 +19,42 @@ export function LemonButton({
     className,
     ...buttonProps
 }: LemonButtonProps): JSX.Element {
-    return (
+    const { onClick, ...buttonPropsInternal } = buttonProps
+    let onClickInternal: LemonRowProps<'button'>['onClick'] | undefined
+    let link: string | undefined
+    if (typeof onClick === 'string') {
```
Comment (paolodamico) [inline frontend/src/lib/components/LemonButton/index.tsx:25]: Nit: this seems a tiny bit convoluted, making a suggestion below

File: frontend/src/lib/components/LemonButton/index.tsx [hunk 2]
```diff
@@ -15,15 +19,42 @@ export function LemonButton({
...
+        onClickInternal = onClick
+    }
+    const row = (
         <LemonRow
             tag="button"
             className={clsx('LemonButton', type !== 'default' && `LemonButton--${type}`, className)}
             icon={icon}
             type="button"
-            {...buttonProps}
+            onClick={onClickInternal}
```
Comment (paolodamico) [inline frontend/src/lib/components/LemonButton/index.tsx:36]:
```suggestion
            onClick={typeof onClick !== 'string' ? onClickInternal : undefined}
```

File: frontend/src/lib/components/LemonButton/index.tsx [hunk 3]
```diff
@@ -15,15 +19,42 @@ export function LemonButton({
...
+            {...buttonPropsInternal}
         >
             {children}
         </LemonRow>
     )
+    return link ? <Link to={link}>{row}</Link> : row
```
Comment (paolodamico) [inline frontend/src/lib/components/LemonButton/index.tsx:42]:
```suggestion
    return typeof onClick === 'string' ? <Link to={onClick}>{row}</Link> : row
```

File: frontend/src/lib/components/LemonButton/index.tsx [hunk 4]
```diff
@@ -15,15 +19,42 @@ export function LemonButton({
     className,
     ...buttonProps
 }: LemonButtonProps): JSX.Element {
-    return (
+    const { onClick, ...buttonPropsInternal } = buttonProps
```
Comment (paolodamico) [inline frontend/src/lib/components/LemonButton/index.tsx:22]: Why do we expand the extra properties twice? Worth just passing `buttonProps` down

**Review by paolodamico — APPROVED** [review]
lgtm! only thing that seems off (non-blocking) is that when you're on the new insight page, the insight tab is not highlighted (whether you click on the new plus icon or not)

**Comment by Twixes** [comment]
Good catches.
The slightly convoluted code was left from a previous approach, simplified.

**Pull request merged**